rust/worker/src/server.rs

```diff
@@ -182,32 +184,85 @@ impl WorkerServer {
             }
         };
 
-        let result = match self.system {
-            Some(ref system) => {
-                let orchestrator = HnswQueryOrchestrator::new(
-                    // TODO: Should not have to clone query vectors here
-                    system.clone(),
-                    query_vectors.clone(),
-                    request.k,
-                    request.allowed_ids,
-                    request.include_embeddings,
-                    segment_uuid,
-                    collection_uuid,
-                    self.log.clone(),
-                    self.sysdb.clone(),
-                    self.hnsw_index_provider.clone(),
-                    self.blockfile_provider.clone(),
-                    dispatcher,
-                    collection_version,
-                    log_position,
-                );
-                orchestrator.run().await
-            }
+        let system = match self.system {
+            Some(ref system) => system,
             None => {
+                tracing::error!("No system found");
                 return Err(Status::internal("No system found"));
             }
         };
 
+        let knn_filter_orchestrator = KnnFilterOrchestrator::new(
+            dispatcher.clone(),
+            // TODO: Load the configuration for this
+            1000,
+            FetchLogOperator {
+                log_client: self.log.clone(),
+                batch_size: 100,
+                skip: log_position as u32 + 1,
```

Comment:
nit: if `skip` in practice always accepts a log position maybe it should be called `start_at`

Reply:
Renamed for better clearity